packages/synthetic-chain/src/lib/commonUpgradeHelpers.ts

```diff
@@ -187,13 +207,27 @@ export const voteLatestProposalAndWait = async (
   };
   if (title) {
     proposals = proposals.filter(proposal => {
-      if (proposal.content) {
-        return proposal.content.title === title;
-      } else if (proposal.messages) {
+      if (proposal.title === title) {
+        return true;
+      }
+      if (proposal.content?.title === title) {
+        return true;
+      }
+      if (proposal.messages) {
         return proposal.messages.some(message => {
-          message['@type'] === '/cosmos.gov.v1.MsgExecLegacyContent' ||
-            Fail`Unsupported proposal message type ${message['@type']}`;
-          return message.content.title === title;
+          let typeUrl: string;
+          let msgTitle: string;
+          if ('@type' in message) {
+            typeUrl = message['@type'];
+            msgTitle = message?.content?.title;
+          } else {
+            typeUrl = message.type;
+            msgTitle = message?.value?.content?.value?.title;
+          }
+
```

Comment:
please extract a helper for normalizing the proposal info. there should be no `let`